src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Claims/WindowsClaimSet.cs

```diff
@@ -189,13 +256,13 @@ public override IEnumerable<Claim> FindClaims(string claimType, string right)
                 {
                     if (right == null || Rights.Identity == right)
                     {
-                        yield return new Claim(ClaimTypes.Sid, _windowsIdentity.User, Rights.Identity);
+                        yield return new Claim(ClaimTypes.Sid, _windowsIdentity, Rights.Identity);
                     }
```

Comment:
claimType of Sid is only valid if it's a WindowsIdentity object. If a ClaimTypes.Sid is requested, this will return the wrong thing. You need to make the SupportedClaimType return false is it's not a WindowsIdentity and requesting Sid or DenyOnlySid. Then in this code it needs to be cast as WindowsIdentity and fetch the User property which is the Sid.